packer/ansible/roles/isc-dhcp-server/files/config_isc-dhcp-server.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+##########################################################
+# retrieve NIC name via ```ip addr``` command, by given id
+##########################################################
+get_nic_name_by_index()
+{
+    #######################
+    # This script is to obtain and echo the NIC name (lo/eth0/eth1) by index(1/2/3) shown in ```ip addr```
+    #
+    #
+    # ip addr will show:
+    #
+    #1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default
+    #.......
+    #2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 ...
+    #.......
+    #3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pf....
+    #......
+    #
+    ########################
+    tmp=$( ip addr|grep "^${1}:" | awk '{print $2}')
+    if [[ "$tmp" == "" ]];
+    then
+        echo "[Error]: Invalid Parameter passed to get_nic_name_by_index(): NIC ID=$1";
```

Comment:
This sounds like verifying index parameter passed to ```get_nic_name_by_index```, but in a little bit tricky way. 

I suggest the best place to verify user input is before L23, if index is supposed to be a number, verify it against a number. 

Error catched here by should either be:
1. That's no interface number matching index parameter, maybe the index asked is to large.
2. Unfortunately (happens very rare), ```ip addr``` has changed it's output format thus the 2'nd awk field is no longer a valid interface name.